react-native-scripts/src/util/packager.js

```diff
@@ -61,6 +62,40 @@ function run(onReady: () => ?any, options: Object = {}, isInteractive = false) {
   let progressBar;
   const projectDir = process.cwd();
 
+  if (process.platform !== 'win32') {
+    if (process.platform === 'darwin') {
+      const watcherDetails = spawn.sync('sysctl', ['kern.maxfiles']).stdout.toString();
+      if (parseInt(watcherDetails.split(':')[1].trim()) < 5242880) {
+        log.withTimestamp(
+          `
+The number of files that can be watched is too low. Please set it to higher number.
+Please use:
+${chalk.cyan(`  sudo sysctl -w kern.maxfiles=5242880
+  sudo sysctl -w kern.maxfilesperproc=524288`)}
+and start again.
+        `
+        );
+        process.exit(1);
+      }
+    } else {
+      const watcherDetails = spawn
+        .sync('sysctl', ['fs.inotify.max_user_watches'])
+        .stdout.toString();
+      if (parseInt(watcherDetails.split('=')[1].trim()) < 12288) {
```

Comment:
Hey, thanks for the input.
For those magical numbers I looked at watchman docs and got [these](https://facebook.github.io/watchman/docs/install.html#mac-os-file-descriptor-limits). And I as was not sure about actual numbers for any of those I just used those.

It surely isn't the best solution, so I'll check for this 

> I guess it would be harder to check, but to do this correctly I think you have to detect when the number of watches has reached it's limit, and then print the error.

and see if anythings possible